src/main/java/com/google/devtools/build/lib/remote/blobstore/CombinedDiskHttpBlobStore.java

```diff
@@ -20,27 +20,35 @@
 import com.google.common.util.concurrent.MoreExecutors;
 import com.google.common.util.concurrent.SettableFuture;
 import com.google.devtools.build.lib.vfs.Path;
-import java.io.ByteArrayInputStream;
-import java.io.ByteArrayOutputStream;
+
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.util.UUID;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+
 /**
  * A {@link SimpleBlobStore} implementation combining two blob stores. A local disk blob store and a
  * remote http blob store. If a blob isn't found in the first store, the second store is used, and
  * the blob added to the first. Put puts the blob on both stores.
  */
-public final class CombinedDiskHttpBlobStore extends OnDiskBlobStore {
+public final class CombinedDiskHttpBlobStore implements SimpleBlobStore {
```

Comment:
I don't understand what this buys us. The code is still closely tight to the local disk cache by having copied the move logic in `get`. I couldn't possibly plug in anything except the disk cache there?